haproxy/templates/default/haproxy.cfg.frontend.erb

```diff
@@ -1,33 +1,36 @@
-
+<%
+haproxy_cfg = @node.fetch('haproxy', {})
+opsworks_layers = @node.fetch('opsworks', {}).fetch('layers', {})
+%>
 frontend http-in
   bind *:80
-  <% unless @node.fetch('haproxy',{}).fetch('ssl', 'off') == 'off'
+  <% unless haproxy_cfg.fetch('ssl', 'off') == 'off'
     certificate = "#{@node['ssl-deploy']['directory']}/cert.combined.pem"
   -%>
   bind *:443 ssl crt <%= certificate %>
   reqadd X-Forwarded-Proto:\ https if { ssl_fc }
-  <% unless @node.fetch('haproxy',{}).fetch('hsts',{}).nil? -%>
-  rspadd Strict-Transport-Security:\ max-age=<%=@node.fetch('haproxy',{}).fetch('hsts',{})%> if { ssl_fc }
+  <% unless haproxy_cfg.fetch('hsts',{}).nil? -%>
+  rspadd Strict-Transport-Security:\ max-age=<%=haproxy_cfg.fetch('hsts',{})%> if { ssl_fc }
```

Comment:
@fh When I read this statement, I was wondering too if this ever works.

1. Line 12, it seems to default to an object by default, but we check for nil.
2. Line 13, I would expect this to be a string/number and also not an object.